packages/mermaid/src/diagrams/git/gitGraphRenderer.js

```diff
@@ -338,27 +338,24 @@ const drawCommits = (svg, commits, modifyGraph) => {
 };
 
 /**
- * Detect if there are other commits between commit1's x-position and commit2's x-position on the
- * same branch as commit2.
+ * Detect if there are commits
+ * between commitA's x-position
+ * and commitB's x-position on the
+ * same branch as commitA
  *
- * @param {any} commit1
- * @param {any} commit2
+ * @param {any} commitA
+ * @param {any} commitB
  * @param allCommits
- * @returns {boolean} If there are commits between commit1's x-position and commit2's x-position
+ * @returns {boolean}
+ * If there are commits between
+ * commitA's x-position
+ * and commitB's x-position
  */
-const hasOverlappingCommits = (commit1, commit2, allCommits) => {
-  // Find commits on the same branch as commit2
-  const keys = Object.keys(allCommits);
-  const overlappingComits = keys.filter((key) => {
-    return (
-      allCommits[key].branch === commit2.branch &&
-      allCommits[key].seq > commit1.seq &&
-      allCommits[key].seq < commit2.seq
-    );
-  });
-
-  return overlappingComits.length > 0;
-};
+const hasOverlappingCommits = (commitA, commitB, allCommits) =>
+  Object.entries(allCommits).some(
+    (commitX) =>
+      commitX.branch === commitA.branch && commitX.seq > commitA.seq && commitX.seq < commitB.seq
```

Comment:
I think you're right re `Object.entries()`. It should have been `Object.values()`.

Your fix would also work but I don't think we need the key for anything, so grabbing just the values saves us having to make a second look-up (and with any luck, makes the function clearer).

Good catch 👍

Reply:
I've rewritten and renamed this function now so that its purpose should be clearer. Let me know if you spot anything or have any questions or please resolve the conversation if you are happy.